.github/workflows/pypi-release.yml

```diff
@@ -12,14 +15,17 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: '3.x'
+
       - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install setuptools wheel twine
-      - name: Build and publish
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-        run: |
-          python setup.py sdist bdist_wheel
-          twine upload dist/*
+        run: pip install -U build twine
+      - name: Build package
+        run: python -m build
+      - name: Check package
+        run: twine check dist/*
+
+      - name: Publish distribution 📦 to PyPI
```

Comment:
All this action does is calling `twine upload dist/*` and checking inputs. I don't think it's worth adding more complexity here.

Reply:
yes, but then you would need to apply this action on PR and it would fail as you can't push existing version so with respect to reuse the same workflow we need to have it as separate step